validator_test.go

```diff
@@ -301,14 +302,21 @@ func TestValidatorSubDomains(t *testing.T) {
 		},
 	}
 
-	dir, err := ioutil.TempDir("", "emailstest")
-	if err != nil {
-		log.Fatal(err)
-	}
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			g := NewWithT(t)
 
+			// Make a TempFile prefix somehow based on the testcase name
+			dir, err := ioutil.TempDir("", fmt.Sprintf("%s_", tc.name))
```

Comment:
Why a whole `TempDir` instead of just the single `TempFile` you need for the test case?